### PR TITLE
Update release checklist as we've moved to using a CHANGELOG file.

### PR DESCRIPTION
**Pull request by 12wrigja**

### CODE REVIEW

File: release-checklist.md [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
       - `cat CHANGELOG.md | grep -oP '(?<=\[)([a-z0-9]{8})(?=\])' | xargs -I{} bash -c 'echo "[{}]: https://github.com/js-temporal/temporal-polyfill/commit/$(git rev-parse {})"'`
     - It's nice to include links back to the original proposal-temporal PRs when possible. To search through commit messages for relevant links, try `cat CHANGELOG.md | grep -oP '(?<=^\[)([a-z0-9]{8})(?=\])' | xargs -I{} bash -c 'git log --format=%B -n 1 {} | grep "tc39/proposal-temporal" && echo {}'`, which will print out all links **followed by** the relevant commit short ref.
-  - Thank everyone who contributed to the release (e.g. `git shortlog v0.1.0..` to see the contributors)
+- Open a pull request with the above change, get it approved, and merge it into
+  `main`
 - Create the new release on GitHub https://github.com/js-temporal/temporal-polyfill/releases/new
-  - Don't forget the `v` in the version tag, e.g. `v0.2.0`
-  - Check the "This is a pre-release" box unless we've decided to release a production version
-- Run `NODE_ENV=production npm publish` to publish the release to NPM
+  - In the release desdription, include a permalink to the CHANGELOG.md file at the commit created earlier.
```
Comment (gilmoreorless) [inline release-checklist.md:18]: Drive-by comment from a lurking watcher of the project, sorry!
```suggestion
  - In the release description, include a permalink to the CHANGELOG.md file at the commit created earlier.
```

Comment (12wrigja) [reply release-checklist.md]: Thanks! Much appreciated.
